### PR TITLE
make `recompose()` work with sparse tibbles

### PR DESCRIPTION
**Pull request by EmilHvitfeldt**
Ref: https://github.com/tidymodels/hardhat/issues/253

### CODE REVIEW

File: R/sparsevctrs.R [hunk 1]
```diff
@@ -0,0 +1,3 @@
+is_sparse_tibble <- function(x) {
```
Comment (hfrick) [inline R/sparsevctrs.R:1]: is this automatically a tibble? I.e. can sparsevctrs vectors only exist in tibbles or is this actually more `has_sparse_vectors()`?

**Review by hfrick — APPROVED** [review]
Looks good! This is a nice breakdown of isolated tasks in #253. I'm generally very on board with PRs that do one little thing in a bigger picture. Could you, in future, link back to the bigger picture in the PR description? In this case here, I followed the breadcrumb trail of issues and PRs back to workflows where things come together and that helps me put this PR into context and review better.

File: R/sparsevctrs.R (hunk 1 above)
Comment (EmilHvitfeldt) [reply R/sparsevctrs.R:1]: it is. good point. sparse vectors can exists by themselves or in lists, data.frames or tibbles. This function will soon live in {sparsevctrs} itself, properly with a better name

https://github.com/r-lib/sparsevctrs/issues/70

Comment (EmilHvitfeldt) [reply R/sparsevctrs.R:1]: more specifically, `recompose()` when used in {recipes} always takes a tibble as input

**Comment by EmilHvitfeldt** [comment]
> Could you, in future, link back to the bigger picture in the PR description?

sorry about that. Will do!

For future readers, this PR was needed because `recompose()` is used in {recipes} as the function that casts to different `composition`s that are specified in `bake()`

**Pull request merged**